samza-core/src/main/java/org/apache/samza/zk/ZkSessionMetrics.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.zk;
+
+import org.apache.samza.metrics.Counter;
+import org.apache.samza.metrics.MetricsBase;
+import org.apache.samza.metrics.MetricsRegistry;
+
+public class ZkSessionMetrics extends MetricsBase {
+
+  /**
```

Comment:
consider wrapping these in ZkJobCoordinatorMetrics so that all metrics for monitoring Zk are in one place?

Reply:
Unifying would keep all metrics in one place, but ZkSessionMetrics and ZkCoordinatorMetrics are different category of metrics. It would be logical to keep them separate. What do you think?

Reply:
well, they are not entirely different because the coordinator handles Zk-session management, barrier lifecycle etc. but, it's fine by me if you prefer to keep them in 2 separate classes